Test a few React components

robintown opened the pull request:
Since so much of this code base is made up React components, we do need to test them, and I wanted to see how easy it would be to get some automated accessibility tests with [axe](https://github.com/dequelabs/axe-core/) up and running. Along the way I discovered the reason why we were seeing funny errors in our tests (it was because I had turned off Vitest's test isolation option), and the tests I wrote also caught a [real accessibility issue](https://github.com/element-hq/compound-web/pull/246) in Compound Web.

robintown committed:
Test RoomHeaderInfo

robintown committed:
Enable test isolation

I had experimented with turning this off in order to improve test performance, and apparently that ended up being merged. Now if we're to do component testing, we'll be changing things globally on the document, so isolation is very much necessary.

robintown committed:
Test InviteModal

robintown committed:
Test GridTile

robintown committed:
Test SpotlightTile

robintown committed:
Upgrade Compound Web

This patch release fixes a bug where tooltips would label non-interactive triggers in an inaccessible way.

toger5 commented on vitest.config.js:
```diff
@@ -12,7 +12,6 @@ export default defineConfig((configEnv) =>
             classNameStrategy: "non-scoped",
           },
         },
-        isolate: false,
```
That very much explains what I have observed.

toger5 commented on src/Modal.tsx:
```diff
@@ -98,6 +98,7 @@ export const Modal: FC<Props> = ({
               styles.drawer,
               { [styles.tabbed]: tabbed },
             )}
+            aria-describedby={undefined}
```
Why do we need this?

toger5 commented on src/tile/SpotlightTile.test.tsx:
```diff
@@ -0,0 +1,57 @@
+          showIndicators
+        />,
+      );
+      expect(await axe(container)).toHaveNoViolations();
+      // Name should be visible
+      screen.getByText("Alice");
+    },
```
We could add a lot more expectations here:
 - avatar url
 - expanded
 - indicators
 ...

robintown replied:
Right, I've added some more expectations now. I wasn't able to test the specific avatar URL, as that requires a full Matrix client setup, and the indicators are (intentionally, IIRC?) excluded from the accessibility tree. But, there were still some useful things to test, and I actually caught a couple more accessibility issues this way!

toger5 commented on src/utils/test.ts:
```diff
@@ -58,3 +65,67 @@ export function withTestScheduler(
   );
 }
+
+export async function withLocalMedia(
```
Would it make sense to not require tests (ignore test utils in codecov) on our test utils?

robintown committed:
Exclude test utilities from coverage report

robintown committed:
Explain why I've added aria-describedby={undefined}

robintown committed:
Test SpotlightTile more thoroughly

Catching two accessibility issues along the way: we were putting the wrong accessible labels on the 'expand' button, and even the off-screen pages of the spotlight tile were being exposed to accessibility technologies rather than hidden.

robintown committed:
Merge branch 'livekit' into test-components

robintown committed:
Test StarRating
